modules/imgproc/src/demosaicing.cpp

```diff
@@ -1566,9 +1566,9 @@ class Bayer2RGB_EdgeAware_T_Invoker :
             int x = 1;
             if (start_with_green)
             {
-                D[blue<<1] = (S[-sstep] + S[sstep]) >> 1;
+                D[blue<<1] = (S[-sstep] + S[sstep] + 2) >> 1;
                 D[1] = S[0];
-                D[2-(blue<<1)] = (S[-1] + S[1]) >> 1;
+                D[2-(blue<<1)] = (S[-1] + S[1] + 2) >> 1;
```

Comment:
`+ 1` must be here

Reply:
I did not get you as i am not totally familiar with the code base being a new contributor. Can you elaborate on what you mean by +1 ?

Reply:
You should follow the rounding rules. https://en.wikipedia.org/wiki/Rounding

`>> 1` means `* 0.5` (optimized version)
`>> 2` means `* 0.25` (optimized version)

You are switching from `Rounding towards zero` to `Round half up`.

In the second case you adding 0.5 <==> `(x + 2) >> 2`
In the first case you should add 0.5 too ==> `(x + 1) >> 1`

Reply:
oh ok thanks will make the required changes